group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Record.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group;
+
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+
+import java.util.Objects;
+
+/**
+ * A Record which contains an {{@link ApiMessageAndVersion}} as key and
+ * an {{@link ApiMessageAndVersion}} as value. The value could be null.
+ *
```

Comment:
can we add some more description on where this Record object is used? i.e. for the __consumer_offsets topic and that since it's a compacted topic it will always have a key & null value would be a tombstone.

Reply:
Yeah I realized I didn't understand why both are ApiMessageAndVersion and how one maps to the other.

Reply:
Is the general vagueness intentional so we can adapt and use in the transaction coordinator too?

Reply:
Rephrases the comment a bit. I don't want to mention `__consumer_offsets` here because this Record is not attached to it at all.